beacon-chain/core/blocks/block_operations.go

```diff
@@ -250,139 +237,114 @@ func ProcessAttesterSlashings(
+	var slashableIndices []uint64
+	for _, idx1 := range slashableAttestation1.ValidatorIndices {
+		for _, idx2 := range slashableAttestation2.ValidatorIndices {
+			if idx1 == idx2 {
+				if beaconState.ValidatorRegistry[idx1].PenalizedEpoch > helpers.CurrentEpoch(beaconState) {
+					slashableIndices = append(slashableIndices, idx1)
+				}
+			}
+		}
 	}
-	return indicesIntersection, nil
+	// Verify that len(slashable_indices) >= 1.
+	if len(slashableIndices) < 1 {
+		return nil, errors.New("expected a non-empty list of slashable indices")
+	}
+	return slashableIndices, nil
 }
 
-func verifySlashableVote(votes *pb.SlashableVote, verifySignatures bool) error {
-	emptyCustody := make([]byte, len(votes.CustodyBitfield))
-	if bytes.Equal(votes.CustodyBitfield, emptyCustody) {
+func verifySlashableAttestation(att *pb.SlashableAttestation, verifySignatures bool) error {
+	emptyCustody := make([]byte, len(att.CustodyBitfield))
+	if bytes.Equal(att.CustodyBitfield, emptyCustody) {
 		return errors.New("custody bit field can't all be 0s")
 	}
-	if len(votes.ValidatorIndices) == 0 {
+	if len(att.ValidatorIndices) == 0 {
 		return errors.New("empty validator indices")
 	}
-	for i := 0; i < len(votes.ValidatorIndices)-1; i++ {
-		if votes.ValidatorIndices[i] >= votes.ValidatorIndices[i+1] {
+	for i := 0; i < len(att.ValidatorIndices)-1; i++ {
+		if att.ValidatorIndices[i] >= att.ValidatorIndices[i+1] {
 			return fmt.Errorf("validator indices not in descending order: %v",
-				votes.ValidatorIndices)
+				att.ValidatorIndices)
 		}
 	}
-	if len(votes.CustodyBitfield) != mathutil.CeilDiv8(len(votes.ValidatorIndices)) {
+	if len(att.CustodyBitfield) != mathutil.CeilDiv8(len(att.ValidatorIndices)) {
 		return fmt.Errorf("custody bit field length (%d) don't match indices length (%d)",
-			len(votes.CustodyBitfield), mathutil.CeilDiv8(len(votes.ValidatorIndices)))
+			len(att.CustodyBitfield), mathutil.CeilDiv8(len(att.ValidatorIndices)))
 	}
-	if uint64(len(votes.ValidatorIndices)) > params.BeaconConfig().MaxIndicesPerSlashableVote {
+	if uint64(len(att.ValidatorIndices)) > params.BeaconConfig().MaxIndicesPerSlashableVote {
 		return fmt.Errorf("validator indices length (%d) exceeded max indices per slashable vote(%d)",
-			len(votes.ValidatorIndices), params.BeaconConfig().MaxIndicesPerSlashableVote)
+			len(att.ValidatorIndices), params.BeaconConfig().MaxIndicesPerSlashableVote)
 	}
 
 	if verifySignatures {
 		// TODO(#258): Implement BLS verify multiple.
-		//  pubs = aggregate_pubkeys for each validator in registry for poc0 and poc1
-		//    indices
-		//  bls_verify_multiple(
-		//    pubkeys=pubs,
-		//    messages=[
-		//      hash_tree_root(votes)+bytes1(0),
-		//      hash_tree_root(votes)+bytes1(1),
-		//      signature=aggregate_signature
-		//    ]
-		//  )
 		return nil
 	}
 	return nil
 }
 
+func isDoubleVote(data1 *pb.AttestationData, data2 *pb.AttestationData) bool {
```

Comment:
Doesn't a double vote have to come from the same attester? Wouldn't this be better named as `isSameSlot`?

Reply:
def is_double_vote(attestation_data_1: AttestationData,
                   attestation_data_2: AttestationData) -> bool:
    """
    Check if ``attestation_data_1`` and ``attestation_data_2`` have the same target.
    """
    target_epoch_1 = slot_to_epoch(attestation_data_1.slot)
    target_epoch_2 = slot_to_epoch(attestation_data_2.slot)
    return target_epoch_1 == target_epoch_2